src/main/scala/io/qbeast/spark/delta/writer/SparkDeltaDataWriter.scala

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2021 Qbeast Analytics, S.L.
+ */
+package io.qbeast.spark.delta.writer
+
+import io.qbeast.IISeq
+import io.qbeast.core.model._
+import io.qbeast.spark.index.QbeastColumns
+import io.qbeast.spark.index.QbeastColumns.cubeColumnName
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.mapreduce.Job
+import org.apache.spark.qbeast.config.{MAX_FILE_SIZE_COMPACTION, MIN_FILE_SIZE_COMPACTION}
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.delta.actions.{AddFile, FileAction, RemoveFile}
+import org.apache.spark.sql.execution.datasources.OutputWriterFactory
+import org.apache.spark.sql.execution.datasources.parquet.ParquetFileFormat
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.util.{QbeastThreadUtils, SerializableConfiguration}
+import scala.collection.parallel.ForkJoinTaskSupport
+import scala.collection.parallel.immutable.ParVector
+
+/**
+ * Spark implementation of the DataWriter interface.
+ */
+object SparkDeltaDataWriter extends DataWriter[DataFrame, StructType, FileAction] {
+
+  override def write(
+      tableID: QTableID,
+      schema: StructType,
+      qbeastData: DataFrame,
+      tableChanges: TableChanges): IISeq[FileAction] = {
+
+    val sparkSession = qbeastData.sparkSession
+
+    val (factory: OutputWriterFactory, serConf: SerializableConfiguration) = {
```

Comment:
Interesting fragment. Of course it is a matter of taste, however I wonder what are the reasons to compute factory ans serConf as a pair? Could the following code have a better readability:
```scala
val job = Job.getInstance()
val factory = new ParquetFileFormat().prepareWrite(sparkSession, job, Map.empty, schema)
val serConf = new SerializableConfiguration(job.getConfiguration)
```

Reply:
Sure :laughing: 
It was something copied-and-pasted from based methods. But yes, nothing seems to indicate we need that programming style.